### DISCUSSION TITLE
Piano Project Q&A

### DISCUSSION THREAD

**Discussion by bbswitzer** [bbswitzer/PianoProject#5] [Q&A]
This is a place where you can ask questions and get them answered! This could be about any part of the project from the code to the construction of the device. Questions can be answered by others who are working on the project or by me.

**Comment by Agentblr1** [comment]
Hello. Two Arduino boards can be flashed and they work fine. Please help me with the issue of compiling the firmware for ESP32. I have already tried a lot of things (and different versions of Arduino Core ESP32 from 1.0.3 to 2.0.11, and Visual Studio 2019 with Visual Micro, Visual Studio 2022), but the Bluetooth server does not start on the ESP32 board after firmware, the phone does not see the Piano Player point. Maybe there are BIN files for the ESP32 firmware or another version of the program code?

**Comment by halecroft** [comment]
Does it say the code uploaded correctly, and are you trying to search for the Bluetooth connection using your phone's standard Bluetooth menu? As far as I'm aware it doesn't show up there; you need an app that specifically looks for Bluetooth MIDI devices, which it will show up under. If you have access to a computer with Bluetooth, try using that to confirm the Bluetooth is broadcasting.
________________________________

**Comment by Agentblr1** [comment]
Yes, I use a special BLE midi app to search for the device. I have already tried BLE midi examples and other libraries, they create a BLE midi point, everything is connected. But the Bluetooth midi server does not start on this ESP32 code. I also watched it through the port monitor in debugging mode. Only the code starts after restarting with the launch of keys 1 to 88 and that's it.

**Comment by Agentblr1** [comment]
![1709790056160](https://github.com/bbswitzer/PianoProject/assets/160355313/5049eb6c-3bb9-400e-9234-3c5d282cce4c)
